README.md

```diff
@@ -17,62 +17,98 @@ As a part of the bridge, a Kafka producer will be responsible for producing mess
 
-    This rule maps MQTT topics of the form `building/{building}/room/{room}/#` to the Kafka topic `building_{building}_room_{room}`. 
-    For example, if the MQTT topic is `building/A/room/1/floor/2` it will be mapped to the Kafka topic `building_A_room_1_floor_2`.
+2. MQTT Topic: `sensors/([^/]+)/data` -> Kafka Topic: `sensor_data` with Kafka Key: `null`
 
-2. MQTT Topic: `sensors/+/data` -> Kafka Topic: `sensor_data`
+   This rule maps MQTT topics of the form `sensors/+/data` to the Kafka topic `sensor_data`.
+   For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data`.
+   Because the `KafkaKey` is not defined, the key of the Kafka record will be `null`.
 
-    This rule maps MQTT topics of the form `sensors/+/data` to the Kafka topic `sensor_data`. 
-    For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data`.
+3. MQTT Topic: `sensors.*` -> Kafka Topic: `sensor_others` with Kafka Key: `null`
 
-3. MQTT Topic: `sensors/#` -> Kafka Topic: `sensor_others`
+   This rule maps any MQTT topic starting with `sensors/` followed by any number of levels in the hierarchy to the Kafka
+   topic `sensor_others`.
+   For example, if the MQTT topic is `sensors/temperature/living-room`, it will be mapped to the Kafka
+   topic `sensor_others` with the key `sensor`.
+   Because the `KafkaKey` is not defined, the key of the Kafka record will be `null`.
 
-   This rule maps any MQTT topic starting with `sensors/` followed by any number of levels in the hierarchy to the Kafka topic `sensor_others`.
-   For example, if the MQTT topic is `sensors/temperature/living-room`,  it will be mapped to the Kafka topic `sensor_others`.
+4. MQTT Topic: `devices/([^/]+)/data/(\b(all|new|old)\b)` -> Kafka Topic: `device_$1_data` with Kafka Key: `device_$2`
 
+   This rule maps MQTT topics of the form `devices/{some word}/data/{either all, new or old}` to the Kafka
+   topic `device_$1_data`.
+   For example, if the MQTT topic is `devices/thermostat/data/all`, it will be mapped to the Kafka
+   topic `device_thermostat_data` with the key `device_all`.
+   This example also shows the advantage of using regex in the `mqttTopic` pattern. The last capturing group in
+   the `mqttTopic` should be a word boundary `\b` followed by either `all`, `new` or `old` and anything else will not
+   match the pattern.
 
 The order in which the rules are defined is important.
 The MQTT Bridge will use the first rule that matches the MQTT topic.
-For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data` because `sensors/+/data` matches the MQTT topic before `sensors/#`. 
-If we swap the positions of the rules, the MQTT Bridge would use the `sensors/#` to map the MQTT topic to the Kafka topic `sensor_others`.
+For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data`
```

Comment:
```suggestion
For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data` because `sensors/([^/]+)/data` matches the MQTT topic before `sensors/#`.
```